contrib/platform/src/com/sun/jna/platform/win32/WinDef.java

```diff
@@ -1774,4 +1774,54 @@ public HGLRCByReference(HGLRC h) {
             super(h);
         }
     }
+
+    /**
+     * COORD structure
+     */
+    @FieldOrder({ "X", "Y" })
+    public static class COORD extends Structure {
+
+        public short X;
+        public short Y;
+
+        @Override
+        public String toString() {
+            return String.format("COORD[%s,%s]", X, Y);
```

Comment:
X and Y are primitives, so should the `%s` be `%d` here?

Reply:
When I wrote this, I thought surely Java was more efficient knowing it had an integer value here.  In fact, the way you have it simply invokes `Short.toString(X)` while the `%d` version goes through a bunch of conversions to `BigInteger`, extracting a `long` from that, and finally calling `Long.toString()`, although I don't know if any of those steps are eventually optimized out...

Reply:
My personal rule is to avoid `%d` because you will unexpectedly get different results depending on `Locale.getDefault()`.

e.g.
```
Locale.setDefault(new Locale("ar"))
def s = String.format("%d,%d,%d", 1, 2, 3)
System.out.println(s)
// prints "١,٢,٣" (and not "1,2,3" as one would expect)
```

Reply:
Don't ask me how I know this. Those are many ours of my life I won't get back. `٣` doesn't match `\d` and can't be parsed as  number, and Windows CMD won't print it because it's not ASCII either.

Reply:
Sounds reasonable!